synapse/events/spamcheck.py

```diff
@@ -125,3 +125,25 @@ def user_may_publish_room(self, userid, room_id):
             return True
 
         return self.spam_checker.user_may_publish_room(userid, room_id)
+
+    def check_for_banned_user(self, userid, display_name):
```

Comment:
I do not really like the name of this method. I wonder if something more specific would be better (maybe `show_user_in_searches`)? Does anyone have a better suggestion here?

Reply:
Thanks, it is easy to change again before merging if we come up with anything better.